evm-adapters/src/sputnik/cheatcodes/cheatcode_handler.rs

```diff
@@ -114,14 +114,13 @@ pub(crate) fn convert_log(log: Log) -> Option<String> {
             format!("{}: 0x{}", inner.key, hex::encode(inner.val))
         }
         LogNamedDecimalIntFilter(inner) => {
-            let val: String = inner.val.to_string();
-            let neg = val.trim().starts_with('-');
-            let val = val.trim().trim_start_matches('-');
+            use ethers::core::types::Sign;
+            let (sign, val) = inner.val.into_sign_and_abs();
             format!(
-                "{}: {}{:?}",
+                "{}: {}{}",
                 inner.key,
-                if neg { "-" } else { "" },
-                ethers::utils::parse_units(val, inner.decimals.as_u32()).unwrap()
+                if sign == Sign::Negative { "-" } else { "" },
```

Comment:
sign should implement Display so it can be passed as is to format!

Reply:
That would be idiomatic yeah. I'll add it in

Reply:
Never mind, it's already upstream 👍